src/main/java/com/github/jcustenborder/kafka/connect/snmp/SnmpTrapSourceTask.java

```diff
@@ -104,22 +105,42 @@ public void start(Map<String, String> settings) {
     this.snmp.addCommandResponder(this);
 
     if (this.config.mpv3Enabled) {
+      log.debug("Setting up Mpv3 with protocols {} and {}", this.config.authenticationProtocol, this.config.privacyProtocol);
       setupMpv3Usm(this.snmp, this.config, securityProtocols);
     }
 
   }
 
 
+  private static List<SourceRecord> drain(int batchSize, RecordBuffer<SourceRecord> records) {
```

Comment:
I think it's better to implement this inside the `RecordBuffer` as the original class did, so you could synchronize it with the rest of the methods. Otherwise, for example, someone clan clear the buffer while this draining is going on with an already determined batch size

Reply:
Moved to be internal and changed to synzhronized. I think it should do the trick without the locking as it polls single element, checks for null and returns best effort. Should have no issue with multiple threads polling, though in this case it is only the one in task that polls this. Only thing that could be done in multi thread is the addition as it is handled by the SNMP listener with n-number of threads (defined in configuration)